Chap_Introduction.tex

```diff
@@ -233,3 +234,39 @@ \subsection{PMIx Roles}
 \adviceimplend
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\subsection{Application Binary Interface (ABI)}
+
+An \acf{API} defines how data types, data structures, and functions are represented in source code.
+An \acf{ABI} defines how data types, data structures, and functions are represented in machine code for a given system.
+An important aspect of an \ac{ABI} is the size, layout, and alignment of data structures.
+A stable \ac{ABI} may allow a program compiled with one implementation of the PMIx Standard to run with a different implementation of the PMIx Standard as long as both implementations adhere to the same \ac{ABI}.
+
+The PMIx Standard strives to maintain a stable \ac{ABI} to support applications and tools that rely on more than one implementation of the PMIx Standard.
+To facilitate such interoperability the PMIx \ac{ASC} maintains a set of standardized headers that are versioned with the PMIx Standard that applications and tools can reference~\footnote{PMIx Headers for ABI Compatibility \url{https://github.com/pmix/pmix-abi}}.
+
+In recognition that there are circumstances where the \ac{ABI} needs to be modified this section defines some guidance for making such modifications.
+Additions to the PMIx interface can occur without breaking \ac{ABI} compatibility.
+Deprecating and removing portions of the PMIx interface does break \ac{ABI} compatibility.
+Modifications to the existing PMIx interface does break \ac{ABI} compatibility.
+Any changes to the \ac{ABI} will occur in a major release, and will be noted in the release notes.
+
+The PMIx \ac{ABI} is comprised of the following:
+\begin{compactalphaenum}
+\item Function signatures
+\item Constants and their values
+\item Structures and their membership (including field position)
+\item Attributes that are required to be implemented and their string representation
+\end{compactalphaenum}
+
+When adding a field to an existing data structure that has padding then that padding must be adjusted so as to preserve the stability of the \ac{ABI}.
+If a data structure must be changed in a way that breaks the \ac{ABI} the authors should consider creating a new data structure if possible.
+Additionally, the authors must take care to update or duplicate existing PMIx interfaces that are impacted by the change.
```

Comment:
Remove. No padding in the current standard. May add in the future if padding in a structure is added.